src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -1072,40 +1072,38 @@ void RNN::prepareParams() {
                                                        key.wDescs,
                                                        key.attr);
 
-        return std::make_shared<DnnlExecutor>(descPtr);
+        return descPtr ? std::make_shared<RnnDnnlExecutor>(descPtr) : nullptr;
```

Comment:
Could it descPtr be NULL?

Reply:
Yes. Then `nullptr` value explicitly indicates that the builder has failed. This is a part of the multi cache usage protocol. Then the value is checked on L1083. The previous implementation always created an executor notwithstanding whether the primitive is empty or not, which might lead to a nullptr dereferencing, since the L1083 always passed.